python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -310,6 +311,38 @@ def create_node(self, node_config, tags, count) -> Dict[str, Any]:
         all_created_nodes.update(created_nodes_dict)
         return all_created_nodes
 
+    @staticmethod
+    def _merge_tag_specs(tag_specs: List[Dict[str, Any]],
+                         user_tag_specs: List[Dict[str, Any]]) -> None:
+        """
+        Merges user-provided node config tag specifications into a base
+        list of node provider tag specifications. The base list of
+        node provider tag specs is modified in-place.
+
+        This allows users to add tags and override values of existing
+        tags with their own, and only applies to the resource type
+        "instance". All other resource types are appended to the list of
+        tag specs.
+
+        Args:
+            tag_specs (List[Dict[str, Any]]): base node provider tag specs
+            user_tag_specs (List[Dict[str, Any]]): user's node config tag specs
+        """
+
+        for user_tag_spec in user_tag_specs:
```

Comment:
Thanks for making this a separate function!
Modification in place seems fine here :)